src/libstd/sys/windows/net.rs

```diff
@@ -97,12 +97,26 @@ impl Socket {
         };
         let socket = unsafe {
             match c::WSASocketW(fam, ty, 0, ptr::null_mut(), 0,
-                                c::WSA_FLAG_OVERLAPPED) {
-                c::INVALID_SOCKET => Err(last_error()),
+                                c::WSA_FLAG_OVERLAPPED | c::WSA_FLAG_NO_HANDLE_INHERIT) {
+                c::INVALID_SOCKET => {
+                    match c::WSAGetLastError() {
+                        c::WSAEPROTOTYPE => {
+                            match c::WSASocketW(fam, ty, 0, ptr::null_mut(), 0,
```

Comment:
To confirm, do you know the performance semantics of this on modern platforms? Do modern Windows platforms support passing `WSA_FLAG_NO_INHERIT_HANDLE` and this fallback is only intended for older platforms? Or is this a UWP-vs-not split?

Reply:
To quote you:

>This was added after Windows 7 SP1, so it's not always available.

This flag works fine on relatively modern windows, no uwp needed.